packages/cli/src/index.ts

```diff
@@ -0,0 +1,224 @@
+  );
+
+program.parse();
+
+const isHeadless = Boolean(program.opts().headless);
+
+export const initialize = async () => {
+  const url = program.opts().url;
+  const sitemapURL = program.opts().sitemapUrl;
+
+  // Browser.
+  // @see https://developer.chrome.com/articles/new-headless/
+  const browser = await puppeteer.launch({
+    devtools: true,
+    headless: isHeadless ? 'new' : false,
+  });
+
+  if (url) {
+    // Single URL.
+    let spinner = ora('Analyzing cookies set on first page visit...').start();
+
+    const cookies = await generatePageVisitCookies(new URL(url), browser);
+
+    const cookiesDetails: Array<CookieLogDetails> = [];
+
+    if (cookies) {
+      cookies.forEach((theCookie) => {
+        const cookie = normalizeCookie(theCookie, url);
+
+        if (cookie) {
+          cookiesDetails.push(cookie);
+        }
+      });
+    }
+
+    const csvCookies: string = getCSVbyObject(cookiesDetails);
+
+    spinner.stop();
+    console.log(clc.green('Done analyzing cookies!'));
+
+    spinner = ora('Analyzing technologies used on the page...').start();
+
+    const technologies = await generateTechnology(url);
+
+    const csvTechnologies: string = getCSVbyObject(
+      technologies.map(({ name }) => ({ name }))
+    );
+
+    spinner.stop();
+    console.log(clc.green('Done analyzing technologies!'));
+    console.log(
+      'The following output files were generated in the "out" directory'
+    );
+    console.log('- cookies.csv (Cookie report)');
+    console.log('- technologies. (Technologies report)');
+    console.log('- data.json (Both reports in JSON)');
+
+    await ensureFile('./out/cookies.csv');
+    await writeFile('./out/cookies.csv', csvCookies);
+
+    await ensureFile('./out/technologies.csv');
+    await writeFile('./out/technologies.csv', csvTechnologies);
+
+    await ensureFile('./out/data.json');
+    await writeFile(
+      './out/data.json',
+      JSON.stringify(
+        { cookies: cookiesDetails, technologies: technologies },
+        undefined,
+        2
+      )
+    );
+  } else if (sitemapURL) {
+    const siteMapper = new Sitemapper({
+      url: sitemapURL,
+      timeout: 3000, // 3 seconds
+    });
+
+    let { sites: urls } = await siteMapper.fetch();
+
+    if (urls.length === 0) {
+      throw new Error('Unable to parse sitemap');
+    }
+
+    const countInput: number = parseInt(
+      await promptly.prompt(
+        `Provided sitemap has ${urls.length} sites. Please enter the number of pages you want to analyze:`,
+        (err, val) => {
+          if (
+            (parseInt(val) && parseInt(val) < 0) ||
+            parseInt(val) > urls.length
+          ) {
+            throw new Error('Bad Input');
+          }
+        }
+      )
+    );
+
+    urls = urls.slice(0, countInput);
+
+    const resources = await Promise.all(
+      urls.map(async (_url) => {
+        const cookies = await generatePageVisitCookies(new URL(_url), browser);
```

Comment:
A site may have a huge number of pages, will it be able to process these if the user enters a large number? Should we also show the status of the processing?